app/controllers/account_sign_up_controller.rb

```diff
@@ -1,6 +1,8 @@
 class AccountSignUpController < ProprietorController
   load_and_authorize_resource instance_name: :account, class: 'Account'
 
+  before_action :ensure_permissions!
```

Comment:
You can also move the condition to the action:

```ruby
before_action :ensure_permissions!, if: :admin_only_tenant_creation?
# ...
private
  def ensure_permissions!
    # If configuration says only Admins can create tenants, then require Admin access to perform any actions
    authorize! :read, :admin_dashboard 
  end

  def admin_only_tenant_creation?
    Settings.multitenancy.admin_only_tenant_creation
  end
```

Reply:
Oh, cool. Learned something new with that.  Will update. :)